gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -341,6 +341,15 @@ def self.run_sig(hook_mod, method_name, original_method, declaration_block)
         Signature.new_untyped(method: original_method)
       end
 
+    current_method = signature.method.owner.instance_method(method_name)
+    file_path, line = current_method.source_location
+
+    if !@was_ever_final_names[method_name] && original_method != current_method &&
+       !file_path.match?(%r{sorbet-runtime[^/]*/lib/})
+      raise "The method `#{method_name}` on #{hook_mod} was overwritten before unwrapping signatures " \
```

Comment:
This doesn't have to be an error case, and IMO if it is going to be an error, it should be configurable.

If the Sorbet wrapper is overwritten, then that is not necessarily an error. We just don't want Sorbet to do the unwrapping in that case, since that would result in broken code. As far as the app is concerned, it has the correct method in place and it is working as intended at this point.

For that reason, I'd much prefer it if we skip unwrapping in this case instead of erroring. We could have a special mode where this gets turned into an exception, but that would have to be opt-in, IMO.

Reply:
I agree. As long as Sorbet does the same as Ruby and respects the latest definition of a method, we don't necessarily need to raise. I changed the tests and code to reflect that.